numpyro/distributions/constraints.py

```diff
@@ -189,6 +189,15 @@ def inv(self, y):
     def log_abs_det_jacobian(self, x, y):
         raise NotImplementedError
 
+    def call_with_intermediates(self, x):
+        return self(x), None
+
+    def log_abs_det_jacobian_with_intermediates(self, x, y, intermediates=None):
```

Comment:
Since this function has the same return type as `log_abs_det_jacobian`, my suggestion would be to just have intermediates as an optional kwarg there `log_abs_det_jacobian(x, y, intermediates=None)`, instead of defining a new method. If the transforms has a default `call_with_intermediates`, the `intermeadiates=None` kwarg will work just fine.

Reply:
Thanks! I just update it.